scripts/extensions/videoEditor/src/VideoEditor.tsx

```diff
@@ -0,0 +1,683 @@
+            },
+            () => {
+                this.setState({
+                    scale: this.getScale(),
+                });
+            },
+        );
+    }
+
+    handleSave = () => {
+        const {dataApi} = this.props.superdesk;
+        const crop = this.getCropRotate(pick(this.state.crop, ['x', 'y', 'width', 'height']));
+        const body = {
+            crop: Object.values(crop).join(','),
+            rotate: this.state.degree,
+            trim: Object.values(this.state.trim).join(','),
+            scale: this.state.quality,
+        };
+
+        const {gettext} = this.props.superdesk.localization;
+
+        if (body.crop === '0,0,0,0') {
+            delete body.crop;
+        }
+        if (body.rotate === 0) {
+            delete body.rotate;
+        }
+        if (body.trim === `0,${this.videoRef.current!.duration}`) {
+            delete body.trim;
+        }
+        if (body.scale === 0) {
+            delete body.scale;
+        }
+        if (!isEmpty(body)) {
+            dataApi
+                .create('video_edit', {
+                    edit: body,
+                    item: this.state.article,
+                })
+                .then(() => {
+                    this.setState({isDirty: false});
+                    this.handleToggleLoading(true, gettext('Video is editing, please wait...'));
+                    this.intervalVideoEdit = window.setInterval(() => {
+                        this.props.superdesk.dataApi
+                            .findOne('video_edit', this.state.article._id + `?t=${Math.random()}`)
+                            .then((result: any) => {
+                                if (result.project.processing.video === false) {
+                                    clearInterval(this.intervalVideoEdit);
+                                    this.handleToggleLoading(false);
+                                    this.handleReset();
+                                    this.setState({
+                                        thumbnails: [],
+                                        videoSrc: result.project.url + `?t=${Math.random()}`,
+                                        article: {
+                                            ...this.state.article,
+                                            ...result,
+                                        },
+                                    });
+                                    this.loadTimelineThumbnails();
+                                }
+                            });
+                    }, 3000);
+                })
+                .catch((err: any) => {
+                    this.showErrorMessage(err);
+                    clearInterval(this.intervalVideoEdit);
+                });
+        }
+    }
+
+    checkIsDirty = () => {
+        const state = pick(this.state, ['crop', 'trim', 'degree', 'quality']);
+        // ignore trim.end as initState don't load video duration due to videoRef can be null when component did mount
```

Comment:
By non-`null` video element, do you mean we only render `VideoEditor` component only when its `Wrapper` finished loading and after we got video url? Or should we use callback ref to load video duration during componentDidMount? Could you elaborate more? Thanks
We have tried to use `Wrapper` solution as you suggested, but at the moment, `VideoEditorHeader` is depending on `VideoEditor`, so if we don't render `VideoEditor`, there is nothing left to show, but if we move `Header` outside, we need to pass props around. And actually, we can refactor [handleCheckingVideo](https://github.com/superdesk/superdesk-client-core/pull/3218/files#diff-6a26d285a3bde2bda5233a909812dbe5R147) to reuse it on `handleSave`

Reply:
I was looking at the code again and I think you don't need a wrapper after all. Sorry for giving a wrong direction.

What I would like do is to ensure that video element is not null as early as possible so we don't need the checks later. For example, it would help accepting a non-null video element as an argument in methods instead of using a possibly null instance value. See https://github.com/idsvn/superdesk-client-core/commit/632a9e5eb422e2705d794da440723f58c36751de

Reply:
You won't be able to avoid all null checks, but use this patterns where you can.

Reply:
Got it, thank you, we will update the code accordingly